src/setnpmreg.ps1

```diff
@@ -1,8 +1,31 @@
-If ((Test-Connection art-bobcat.autodesk.com).PingSucceeded){
-    echo adsk reachable
-    New-Item -Path . -Name ".npmrc" -ItemType "file" -Value "registry=https://npm.autodesk.com/artifactory/api/npm/autodesk-npm-virtual/" -Force
+# This script checks if the adsk network is reachable
+# and creates .npmrc file with appropriate value for the npm registry
+
+$npmRegistry = "https://registry.npmjs.org"
+$adskNpmRegistry = "https://npm.autodesk.com/artifactory/api/npm/autodesk-npm-virtual/"
+
+function createNpmrcFile {
+    param (
+        [Parameter(Mandatory = $true)][string]$registry
+    )
+    Write-Host "Creating .npmrc file with registry=$registry" -ForegroundColor Blue
+    New-Item -Path . -Name ".npmrc" -ItemType "file" -Value "registry=$registry" -Force
 }
-else{
-    echo adsk not reachable
-    New-Item -Path . -Name ".npmrc" -ItemType "file" -Value "registry=https://registry.npmjs.org" -Force
+
+try {
+    Write-Host "Checking if adsk npm registry is reachable..." -ForegroundColor Blue
+    $response = Invoke-WebRequest -Uri $adskNpmRegistry -TimeoutSec 20 -ErrorAction Stop
+
+    if ($response.StatusCode -eq 200) {
+        Write-Host "adsk npm registry is reachable" -ForegroundColor Green
+        createNpmrcFile -registry $adskNpmRegistry
+    }
+    else {
+        Write-Host "adsk npm registry is not reachable" -ForegroundColor Red
+        createNpmrcFile -registry $npmRegistry
+    }
+}
+catch {
+    Write-Host "adsk npm registry is not reachable" -ForegroundColor Red
```

Comment:
maybe add, "adsk npm registry is not reachable, falling back to public registry"?

Reply:
There is some logging in the `createNpmrcFile` method
It will look like
```
powershell.exe -ExecutionPolicy Bypass -File "D:\a\Dynamo\Dynamo\Dynamo\src\setnpmreg.ps1"
Checking if adsk npm registry is reachable...
adsk npm registry is not reachable
Creating .npmrc file with registry=https://registry.npmjs.org/
```